src/Symfony/Component/Form/FormRegistry.php

```diff
@@ -97,13 +98,33 @@ public function getType($name)
      */
     private function resolveAndAddType(FormTypeInterface $type)
     {
+        static $checkedTypes = array();
```

Comment:
does this need to be static, or can it be a regular private property?
also, what about replacing the `$checkedTypes = array();` reset by a finally?
That'd be more robust I think

Reply:
Initially when I had it as a private property, it didn't work and still ended in an infinite loop, but I must have been doing something wrong since it's working fine now :)
Also this still needs to support PHP 5.3, so we can't use finally